modules/top-level/plugins/default.nix

```diff
@@ -0,0 +1,52 @@
+/*
+  Creates the final list of (normalized) plugins that will be passed to the wrapper.
+
+  It notably implements:
+  - byte-compilation (performance.byteCompileLua): in this file
+  - plugins combining (performance.combinePlugins): in ./combine-plugins.nix
+*/
+{
+  lib,
+  config,
+  pkgs,
+  builders,
+  ...
+}:
+let
+  inherit (import ./utils.nix) normalizePluginList;
+  inherit (import ./combine-plugins.nix { inherit lib pkgs; }) combinePlugins;
+
+  # Byte compiling of normalized plugin list
+  byteCompilePlugins =
+    plugins:
+    let
+      byteCompile =
+        p:
+        (builders.byteCompileLuaDrv p).overrideAttrs (
+          prev: lib.optionalAttrs (prev ? dependencies) { dependencies = map byteCompile prev.dependencies; }
+        );
+    in
+    map (p: p // { plugin = byteCompile p.plugin; }) plugins;
+
+  # Normalized and optionally byte compiled plugin list
+  normalizedPlugins =
+    let
+      normalized = normalizePluginList config.extraPlugins;
+    in
+    if config.performance.byteCompileLua.enable && config.performance.byteCompileLua.plugins then
+      byteCompilePlugins normalized
+    else
+      normalized;
+
+  combinedPlugins = combinePlugins {
+    inherit (config.performance.combinePlugins)
+      standalonePlugins
+      pathsToLink
+      ;
+    inherit normalizedPlugins;
+  };
+
+  # Plugins to use in build.package
+  plugins = if config.performance.combinePlugins.enable then combinedPlugins else normalizedPlugins;
+in
+plugins
```

Comment:
Instead of _returning_ `plugins`, we could turn this into a module and _define_ an internal `build.normalisedPlugins` option (or some other name)

```nix
{
  options.build.normalisedPlugins = mkOption {
    type = listOf package;
    internal = true;
    # ...
  };

  config.build.normalisedPlugins = if config.performance.combinePlugins.enable then combinedPlugins else normalizedPlugins;
}
```